clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1043,6 +1078,10 @@ class SyclKernelDeclCreator
   size_t LastParamIndex = 0;
 
   void addParam(const FieldDecl *FD, QualType FieldTy) {
+    const ConstantArrayType *CAT =
```

Comment:
This logic here seems a little odd... I wonder if this function needs to be overloaded or refactored to work correctly for both cases.

Reply:
Unlike struct fields, elements of an array have no identity of their own. This addParam call occurs for both scalar fields and for array elements that are scalars. Therefore the check whether this scalar is being expanded as part of an array.